### PR TITLE
BREAKING: WithErrorHandler server option to engine option

### PR DESCRIPTION
**Pull request by dylanhitt**
Moves WithErrorHandler to an engine option.

Closed #314 in favor of this as I created a mess over there.

### CODE REVIEW

File: engine_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+			WithErrorHandler(func(err error) error {
+				return errors.New("")
+			}),
+		)
+		err := NotFoundError{
+			Err: errors.New("Not Found :c"),
+		}
+		errResponse := e.ErrorHandler(err)
+		require.NotErrorAs(t, errResponse, &HTTPError{})
+	})
```
Comment (ccoVeille) [inline engine_test.go:30]:
```suggestion
	t.Run("custom handler", func(t *testing.T) {
		errSentinel := errors.New("foo bar")
		e := NewEngine(
			WithErrorHandler(func(err error) error {
				return errSentinel
			}),
		)
		err := NotFoundError{
			Err: errors.New("Not Found :c"),
		}
		errResponse := e.ErrorHandler(err)
		require.NotErrorAs(t, errResponse, &HTTPError{})
		require.ErrorIs(t, errResponse, errSentinel)
	})
```

File: engine_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		}
+		errResponse := e.ErrorHandler(err)
+		require.ErrorAs(t, errResponse, &HTTPError{})
+	})
+	t.Run("custom handler", func(t *testing.T) {
+		e := NewEngine(
+			WithErrorHandler(func(err error) error {
+				return errors.New("")
+			}),
+		)
```
Comment (ccoVeille) [inline engine_test.go:24]: Please add a test about a custom error handler that could be used to invalidate an error by returning nil

File: engine_test.go (hunk 1 above)
Comment (ccoVeille) [inline engine_test.go:30]: Please add a test about a custom wrapper that wrap the initial one

```suggestion
	t.Run("custom handler", func(t *testing.T) {
		e := NewEngine(
			WithErrorHandler(func(err error) error {
				return fmt.Errorf("%w foobar", err)
			}),
		)
		err := NotFoundError{
			Err: errors.New("Not Found :c"),
		}
		errResponse := e.ErrorHandler(err)
		require.ErrorAs(t, errResponse, &HTTPError{})
		require.ErrorIs(t, errResponse, err)
		require.ErrorContains(t, errResponse, "foobar")
	})
```

File: engine_test.go (hunk 2 above)
Comment (dylanhitt) [reply engine_test.go:24]: Why?

File: engine_test.go (hunk 1 above)
Comment (dylanhitt) [reply engine_test.go:30]: ermmm okay sure

File: engine_test.go (hunk 2 above)
Comment (ccoVeille) [reply engine_test.go:24]: Because people may do that in their custom handler if you dont want to support it, the mechanism of error handling should be rewritten to forbid this behavior

File: engine_test.go (hunk 1 above)
Comment (dylanhitt) [reply engine_test.go:30]: You suggestion doesn't pass.

File: engine_test.go (hunk 2 above)
Comment (dylanhitt) [reply engine_test.go:24]: That is not the responsibility of the function being tested. That would need to be tested [here](https://github.com/go-fuego/fuego/blob/2d6cfa42ba8daa774d1c34b7beea66c0772b478d/serve.go#L98) to give it any sort of meaning.

File: engine_test.go (hunk 1 above)
Comment (ccoVeille) [reply engine_test.go:30]: That the custom error handler is receiving the error and is able to manipulate it.

File: engine_test.go (hunk 2 above)
Comment (dylanhitt) [reply engine_test.go:24]: You're totally right. But I'm not adding this functionality. I'm simply moving the function. I'd prefer an issue is opened to cover this case rather than it be within this PR as that is not the spirit of this unit of work.

File: engine_test.go (hunk 1 above)
Comment (dylanhitt) [reply engine_test.go:30]: Okay sure. I'll update the suggestion to work. fmt.ErrorF actually creates `wrapError` which is why the above doesn't work

Comment (dylanhitt) [reply engine_test.go]: Updated

File: engine_test.go (hunk 2 above)
Comment (ccoVeille) [reply engine_test.go]: Yes, of course. I can understand that.

Let me open an issue

**Review by ccoVeille — APPROVED** [review]

**Review by EwenQuim — APPROVED** [review]
